### PR TITLE
Optimize simple hint shadow algorithm

### PR DESCRIPTION
**Pull request by soulasuna**
## Optimize simple hint shadow algorithm

Changes proposed in this pull request:
- Optimize simple hint shadow algorithm

### CODE REVIEW

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/algorithm/shadow/hint/SimpleHintShadowAlgorithm.java [hunk 1]
```diff
@@ -53,8 +65,17 @@ public boolean isShadow(final Collection<String> shadowTableNames, final Precise
         if (ShadowOperationType.HINT_MATCH != noteShadowValue.getShadowOperationType() && !shadowTableNames.contains(noteShadowValue.getLogicTableName())) {
             return false;
         }
-        return ShadowHintExtractor.extractSimpleHint(noteShadowValue.getValue())
-                .filter(optional -> props.entrySet().stream().allMatch(entry -> Objects.equals(entry.getValue(), optional.get(String.valueOf(entry.getKey()))))).isPresent();
+        return ShadowHintExtractor.extractSimpleHint(noteShadowValue.getValue()).filter(this::containsHint).isPresent();
+    }
+    
+    private boolean containsHint(final Map<String, String> preciseHint) {
+        for (Map.Entry<String, String> entry : simpleHint.entrySet()) {
```
Comment (terrymanu) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/algorithm/shadow/hint/SimpleHintShadowAlgorithm.java:72]: Please do not use `Map.Entry`, just put Map.Entry into import list

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/algorithm/shadow/hint/SimpleHintShadowAlgorithm.java [hunk 2]
```diff
@@ -53,8 +65,17 @@ public boolean isShadow(final Collection<String> shadowTableNames, final Precise
...
+            if (entry.getValue().equals(preciseHint.get(entry.getKey()))) {
+                continue;
```
Comment (terrymanu) [inline shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/algorithm/shadow/hint/SimpleHintShadowAlgorithm.java:74]: It is better to reverse the judge condition and just return here

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/algorithm/shadow/hint/SimpleHintShadowAlgorithm.java (hunk 1 above)
Comment (soulasuna) [reply shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/algorithm/shadow/hint/SimpleHintShadowAlgorithm.java:72]: I will fix it.

File: shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/algorithm/shadow/hint/SimpleHintShadowAlgorithm.java (hunk 2 above)
Comment (soulasuna) [reply shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/main/java/org/apache/shardingsphere/shadow/algorithm/shadow/hint/SimpleHintShadowAlgorithm.java:74]: I will fix it.

**Review by terrymanu — APPROVED** [review]

**Pull request merged**